WordPress/gutenberg
# De-emphasise pattern filters in inserter

## What?
Moves all filtering into a dropdown and removes from pattern explorer for now. Also adds a sticker header so the menu stays in place while scrolling

## Why?
https://github.com/WordPress/gutenberg/issues/54620

The way we filter patterns and highlight different types needs some more time to refine the designs. In the near term we can move filtering into a more discrete place for power users. 

## How?
Consolidate filtering into single file and move it to the top of the pattern list panel.

## Testing Instructions
1. Open post editor
2. Open the  inserter
3. Click patterns tab and select a category
4. Notice filter icon top right which you can click to reveal filtering options
5. Make sure the filtering options work as expected and the menu and header are sticky when scrolling
6. Check that the sync filters can only been activated when the the User pattern source is selected

## Screen recording

https://github.com/WordPress/gutenberg/assets/3629020/25b8d48d-3987-4a6a-885e-cf37b8cf28e2
